internal/conditions/error_to_message.go

```diff
@@ -0,0 +1,39 @@
+package conditions
+
+import (
+	"fmt"
+
+	"github.com/kyma-project/telemetry-manager/internal/workloadstatus"
+)
+
+type ErrorToMessageConverter struct {
+}
+
+func (etc *ErrorToMessageConverter) Convert(err error) string {
+	if workloadstatus.IsPodIsNotScheduledError(err) {
+		//nolint:errcheck,errorlint //errorAs already checks it
+		podNotScheduled := err.(*workloadstatus.PodIsNotScheduledError)
+		return fmt.Sprintf(podIsNotScheduled, podNotScheduled.Message)
+	}
+
+	if workloadstatus.IsPodIsPendingError(err) {
+		//nolint:errcheck,errorlint  //errorAs already checks it
+		podPending := err.(*workloadstatus.PodIsPendingError)
+		if podPending.Reason == "" {
+			return fmt.Sprintf(podIsPending, podPending.ContainerName, podPending.Message)
+		}
+		return fmt.Sprintf(podIsPending, podPending.ContainerName, podPending.Reason)
+	}
```

Comment:
please use errors.As

it will check for the type and convert the matching error in the chain to the requested type at once